packages/flutter_tools/test/general.shard/ios/devices_test.dart

```diff
@@ -46,6 +47,26 @@ void main() {
       IMobileDevice: () => mockIMobileDevice,
     });
 
+    testUsingContext('throws assertion error if platform is not mac os', () async {
+      when(mockIMobileDevice.isInstalled).thenReturn(true);
```

Comment:
Would imobiledevice be installed on Linux? Is there some series of actions that a user could take to get into this state?

Reply:
@zanderso @jonahwilliams I can say with confidence that within the tool code the only place this gets instantiated is in the `IOSDevices` class, which has `bool get supportsPlatform => platform.isMacOS;`. It's possible there's some other tool implementation that instantiates it, which is why I split this out into its own PR.

Looking at the origin of the platform check, it goes back over 3 years ([here](https://github.com/flutter/flutter/commit/67046f933dce5cb8713ae673a664ee323a267840#diff-4e6a9727ba10772d640b597dbdd61d03R105).

Reply:
good point, I need to re-write this test.

Reply:
Done